src/zm_logger.cpp

```diff
@@ -515,8 +515,9 @@ void Logger::logPrint( bool hex, const char * const filepath, const int line, co
     va_list     argPtr;
     struct timeval  timeVal;
 
-    const char * const file = basename(filepath);
-    
+    char *filecopy = strdup(filepath);
```

Comment:
GNU and POSIX functions have different prototypes, one is accepting char*, another - const char*